### DISCUSSION TITLE
How to configure gitlab-ci-local with internal gitlab instance ?

### DISCUSSION THREAD

**Discussion by ssavrim** [firecow/gitlab-ci-local#1354] [Q&A]
Hi,

I try to use the tool with our internal Gitlab instance, but I encounter some issues.
When I execute the command `gitlab-ci-local --list` It tries to connect to `gitlab.com` even if I set environment variables.

I tried in the `$HOME/.gitlab-ci-local/.env`

```yaml
---
CI_SERVER_URL=https://my-gitlab-server
```

I also tried in  `$HOME/.gitlab-ci-local/variables.yml` without any success.

```yaml
---
global:
  CI_SERVER_URL=https://my-gitlab-server
```
Maybe I missed something.

Thanks for your help.

**Comment by ssavrim** [comment]
To bring some more details I got this kind of error 

Given this example:

```yaml
include:
  - project: group1/project1
    ref: master
    file:
      - ci.yml

[...]
```

```bash
$ gitlab-ci-local --list 
git remote get-url origin didn't provide valid matches
Project include could not be fetched { project: group1/project1, ref: master, file: ci.yml }
Error: Command failed with exit code 2: set -eou pipefail; git archive --remote=ssh://git@gitlab.com:22/group1/project1.git master runners.yml | tar -f - -xC .gitlab-ci-local/includes/gitlab.com/group1/project1/master/
error: RPC failed; HTTP 422 curl 22 The requested URL returned error: 422
fatal: git archive: expected ACK/NAK, got a flush packet
tar: This does not look like a tar archive
tar: Exiting with failure status due to previous errors
```

I would expect that the project includes `group1/project1` refers to `https://my-gitlab-server` instead of `gitlab.com`

**Comment by ssavrim** [comment]
I think I got it:

```bash
$ git remote -v
origin  https://gitlabee.dt.renault.com/group1/project1 (fetch)
origin  https://gitlabee.dt.renault.com/group1/project1 (push)
```

By updating the remote ending with `.git`, It works
```bash
$ git remote -v
origin  https://gitlabee.dt.renault.com/group1/project1.git (fetch)
origin  https://gitlabee.dt.renault.com/group1/project1.git (push)
```

**Comment by firecow** [comment]
HTTP origins ain't know as particularly well supported 

https://github.com/firecow/gitlab-ci-local?tab=readme-ov-file#quirks
